app/src/main/java/com/portfolio/tasky/entry/viewModels/RegisterViewModel.kt

```diff
@@ -40,6 +48,10 @@ class RegisterViewModel  @Inject constructor(
         mutableName.value = nameValidation.isValidName(name)
     }
     fun areFieldsValid() {
-        mutableFieldsValid.value =  mutableEmail.value == true && mutablePassword.value == true && mutableName.value == true
+        mutableFieldsValid.value =  mutableEmail.value == true && mutablePassword.value == true && mutableName.value == true && networkObserver.value == NetworkStatus.Available
```

Comment:
I recommend extracting this on a variable, since it will make it a bit more clearer

Ex:
```
val hasCompletedForm= mutableEmail.value == true && mutablePassword.value == true && mutableName.value == true
mutableFieldsValid.value = hasValidatedForm
```

Reply:
True

Done